_episodes/l2_02_basic_unit_testing.md

```diff
@@ -0,0 +1,285 @@
+---
+title: "Introduction to Unit Testing using GoogleTest"
+teaching: TBC
+exercises: TBC
+questions:
+- "What framework/libraries can be used for unit testing in C++?"
+- "What is GoogleTest?"
+- "How to write unit tests with GoogleTest?"
+objectives:
+- "Understand the basic components of a unit test"
+- "Understand how to write a unit test using GoogleTest"
+- "Understand how to compile and run the tests"
+- "Understand some of the macros available in GoogleTest"
+keypoints:
+- "We learnt about some of the popular testing frameworks for C++."
+- "We learnt the baiscs of GoogleTest."
+- "We learnt how to use GoogleTest to write our tests."
+- "We learnt about anatomy of a unit tests." 
+- "We learnt how to write effective unit tests."
+- "Finally, we saw an example to demonstrate the above concepts."
+---
+
+## 1. Popular frameworks for testing in C++
+In C++, a framework refers to a pre-defined software structure or set of libraries that provides a foundation for developing applications or solving specific types of problems. It typically includes a collection of reusable code, tools, and guidelines that help developers build applications more efficiently. There can be different types of frameworks in C++ which include:
+
+1. Application Framework
+2. Web Framework
+3. Game Development Framework
+4. Testing Framework
+
+Testing frameworks help in writing and executing tests for code validation and quality assurance. They provide structures and tools for creating test cases, running tests, and generating reports. There are a number of popular frameworks that can be used for testing in C++. A few of them are mentioned below:-
+
+- [GoogleTest](https://google.github.io/googletest/)
+- [Boost.Test](https://www.boost.org/doc/libs/1_82_0/libs/test/doc/html/index.html)
+- [CppUtest](http://cpputest.github.io/)
+- [Catch2](https://github.com/catchorg/Catch2)
+- [UT++](https://boost-ext.github.io/ut/)
+- [doctest](https://github.com/doctest/doctest)
+- [CTest](https://cmake.org/cmake/help/latest/manual/ctest.1.html)
+
+In this course, we will use GoogleTest, as it is popular and easy to use, though the concepts are the same as for other testing frameworks.
+
+## 2. A brief about GoogleTest
+
+Simply defined, GoogleTest is a testing framework developed by Google's testing technology team to develop and write C++ tests. GoogleTest offers multiple advantages over other frameworks:-
```

Comment:
I think the first sentence is redundant.

Reply:
I took this line from https://google.github.io/googletest/primer.html. The developers themselves started introduction with this line. It think, it should be fine.

Reply:
It is definitely fine for now, but could perhaps be a bit punchier. I think most readers could guess that GoogleTest is a testing framework developed by Google.

Reply:
This looks ok to me. Sure, it is kind of obvious it has been developed by Google, but I don't think it is a big deal.